components/specialized-parser/terraform/main.tf

```diff
@@ -0,0 +1,175 @@
+# limitations under the License.
+
+# Enable APIs
+# See github.com/terraform-google-modules/terraform-google-project-factory
+# The modules/project_services
+locals {
+  # specification of the alloy db docs of removing the .gserviceaccount.com part: https://cloud.google.com/alloydb/docs/manage-iam-authn#create-user
+  alloydb_username           = replace(module.specialized_parser_account.email, ".gserviceaccount.com", "")
+  alloydb_cluster_name_split = split("/", var.alloydb_cluster)
+  alloydb_cluster_name       = element(local.alloydb_cluster_name_split, length(local.alloydb_cluster_name_split) - 1)
+}
+
+module "project_services" {
+  source                      = "github.com/terraform-google-modules/terraform-google-project-factory.git//modules/project_services?ref=ff00ab5032e7f520eb3961f133966c6ced4fd5ee" # commit hash of version 17.0.0
+  project_id                  = var.project_id
+  disable_services_on_destroy = false
+  disable_dependent_services  = false
+  activate_apis = [
+    # General container build and registry
+    "artifactregistry.googleapis.com",
+    "cloudbuild.googleapis.com",
+    "documentai.googleapis.com",
+    "run.googleapis.com",
+    "compute.googleapis.com",
+  ]
+
+  # Provide more access to the cloudbuild service account
+  activate_api_identities = [{
+    "api" : "cloudbuild.googleapis.com",
+    "roles" : [
+      "roles/run.admin",
+      # Required for Cloud Run to launch as the normal compute service account
+      "roles/iam.serviceAccountUser",
+    ]
+    },
+    {
+      "api" : "pubsub.googleapis.com",
+      # PubSub publish to Cloud Run
+      "roles" : [
+        #"roles/iam.serviceAccountUser",
+        "roles/iam.serviceAccountTokenCreator",
+      ],
+    }
+  ]
+}
+
+module "specialized_parser_account" {
+  source     = "github.com/terraform-google-modules/terraform-google-service-accounts?ref=a11d4127eab9b51ec9c9afdaf51b902cd2c240d9" #commit hash of version 4.0.0
+  project_id = var.project_id
+  prefix     = "eks"
+  names      = [local.service_account_name]
+  project_roles = [
+    "${var.project_id}=>roles/documentai.apiUser",
+    "${var.project_id}=>roles/alloydb.databaseUser",
+    "${var.project_id}=>roles/alloydb.client",
+    "${var.project_id}=>roles/serviceusage.serviceUsageConsumer",
+    "${var.project_id}=>roles/documentai.editor",
+    "${var.project_id}=>roles/bigquery.dataEditor",
+    "${var.project_id}=>roles/bigquery.jobUser",
+    "${var.project_id}=>roles/storage.admin",
+  ]
+  display_name = "Specialized Parser Account"
+  description  = "Account used to run the specialized parser jobs"
+}
+
+resource "google_alloydb_user" "specialized_parser_user" {
+  cluster        = var.alloydb_cluster
+  user_id        = local.alloydb_username
+  user_type      = "ALLOYDB_IAM_USER"
+  database_roles = ["alloydbiamuser", "alloydbsuperuser"]
+}
+
+# See github.com/terraform-google-modules/terraform-google-gcloud
+module "set_superuser" {
```

Comment:
nit: for convention with gcloud modules, how about calling this "gcloud_set_superuser".

I find it useful to give the gcloud modules a common prefix because they have different auth and troubleshooting behavior (because it's based on local-exec and runs in a different sub-shell than the rest of Terraform) so it's useful to be able to find them consistently.